From 9b0485c53fffdb27d85c1790abadd790b498040d Mon Sep 17 00:00:00 2001
From: Sasha Levin <sashal@kernel.org>
Date: Wed, 30 Jun 2021 06:16:58 +0200
Subject: mmc: core: Only print retune error when we don't check for card
 removal

From: Wolfram Sang <wsa+renesas@sang-engineering.com>

[ Upstream commit 8335928849729f8e5f10c1497c67260742f7a8cb ]

Skip printing a retune error when we scan for a removed card because we
then expect a failed command.

Signed-off-by: Wolfram Sang <wsa+renesas@sang-engineering.com>
Acked-by: Adrian Hunter <adrian.hunter@intel.com>
Link: https://lore.kernel.org/r/20210630041658.7574-1-wsa+renesas@sang-engineering.com
[Ulf: Rebased patch]
Signed-off-by: Ulf Hansson <ulf.hansson@linaro.org>
Stable-dep-of: b3855668d98c ("mmc: sdhci: Add support for "Tuning Error" interrupts")
Signed-off-by: Sasha Levin <sashal@kernel.org>
---
 drivers/mmc/core/core.c | 12 +++++++-----
 1 file changed, 7 insertions(+), 5 deletions(-)

diff --git a/drivers/mmc/core/core.c b/drivers/mmc/core/core.c
index d8b38203f0fe9..2a8c7d5631d14 100644
--- a/drivers/mmc/core/core.c
+++ b/drivers/mmc/core/core.c
@@ -940,15 +940,17 @@ int mmc_execute_tuning(struct mmc_card *card)
 		opcode = MMC_SEND_TUNING_BLOCK;
 
 	err = host->ops->execute_tuning(host, opcode);
-
-	if (err) {
-		pr_err("%s: tuning execution failed: %d\n",
-			mmc_hostname(host), err);
-	} else {
+	if (!err) {
 		mmc_retune_clear(host);
 		mmc_retune_enable(host);
+		return 0;
 	}
 
+	/* Only print error when we don't check for card removal */
+	if (!host->detect_change)
+		pr_err("%s: tuning execution failed: %d\n",
+			mmc_hostname(host), err);
+
 	return err;
 }
 
-- 
2.43.0

